### PR TITLE
Add load balancing test

### PR DESCRIPTION
**Pull request by KZhu-ME**
Modifies the embedded_circle test to use load balancing, and thus should be merged after https://github.com/ECP-WarpX/WarpX/pull/2538.

This PR/test is intended to fail until after load balancing is fixed for 2D embedded boundaries, so it should not be merged until https://github.com/ECP-WarpX/WarpX/issues/2554 is fixed.

### CODE REVIEW

File: Examples/Tests/embedded_circle/inputs_2d [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+max_step = 11
+warpx.const_dt = 3.99e-13
+warpx.do_electrostatic = labframe
+warpx.self_fields_required_precision = 1e-06
+warpx.eb_implicit_function = -((x-0.00005)**2+(z-0.00005)**2-1e-05**2)
+warpx.eb_potential(x,y,z,t) = -10
+warpx.self_fields_absolute_tolerance = 0.02
+
+algo.load_balance_intervals = 5
+algo.load_balance_efficiency_ratio_threshold = 1.001
```
Comment (roelof-groenewald) [inline Examples/Tests/embedded_circle/inputs_2d:15]:
```suggestion
algo.load_balance_costs_update = timers
algo.load_balance_efficiency_ratio_threshold = 1.001
```

**Review by roelof-groenewald — APPROVED** [review]

File: Examples/Tests/embedded_circle/analysis.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import sys
+sys.path.insert(1, '../../../../warpx/Regression/Checksum/')
+import checksumAPI
+
+# this will be the name of the plot file
+fn = sys.argv[1]
+
+# Get name of the test
+test_name = fn[:-9] # Could also be os.path.split(os.getcwd())[1]
```
Comment (ax3l) [inline Examples/Tests/embedded_circle/analysis.py]: Please use the syntax suggested in the comment :)
Reason: #2385

**Comment by KZhu-ME** [comment]
There still seems to be an issue similar to https://github.com/ECP-WarpX/WarpX/issues/2390. When the test is run in debug mode, the program errors out with message 
`amrex::Abort::1:: (46,62,0,0) is out of bound (47:64,63:80,0:0,0:0) !!!
SIGABRT
amrex::Abort::0:: (46,46,0,0) is out of bound (47:64,47:64,0:0,0:0) !!!
SIGABRT`
The backtrace points to `WarpX::ComputeEdgeLengths()` in `WarpXInitEB.cpp`.

**Comment by KZhu-ME** [comment]
The same error occurs even after https://github.com/ECP-WarpX/WarpX/pull/2607 was merged.

**Comment by PhilMiller** [comment]
Try with #2635

**Comment by KZhu-ME** [comment]
This test passes both manually running it in debug mode and running it through `WarpX/run_test.sh` with https://github.com/ECP-WarpX/WarpX/pull/2635!

**Comment by ax3l** [comment]
@KZhu-ME can you please rebase or merge `development` once more? :)

**Review by ax3l — APPROVED** [review]

**Pull request merged**